confidant/public/modules/history/controllers/BlindCredentialHistoryCtrl.js

```diff
@@ -69,6 +69,14 @@
                 }
             });
 
+            $scope.getCredentialByID = function(id) {
+                return $filter('filter')($scope.$parent.credentialList, {'id': id})[0];
```

Comment:
apologies, by JS is rusty.  We can always expect an entry to be returned by this?  Meaning, `[0]` is safe to return?

Reply:
`$scope.$parent.credentialList` is a pre-generated list. If there's no credentials in that list, then nothing would ever call `getCredentialByID`. If `getCredentialByID`, then something must be in the list.